### PR TITLE
Rename network audit allow list data.

### PR DESCRIPTION
**Pull request by rillian**
Replace the outdated 'whitelist' term with more inclusive language.
These data structures and their references were recently refactored
to move the `network-audit` script into a browsertest, but the names
weren't updated. This changes them to match current style.

Resolves brave/brave-browser#19357

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`npm run test-security`

### CODE REVIEW

**Review by mariospr — APPROVED** [review]
LGTM with a nit: could we rename the (too generic IMHO) `browser/net/brave_network_audit_lists.h` name into `browser/net/brave_network_audit_allowed_lists.h`? 

Thanks!

**Review by iefremov — APPROVED** [review]

File: browser/net/brave_network_audit_allowed_lists.h [hunk 1]
```diff
@@ -3,20 +3,20 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#ifndef BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_WHITELISTS_H_
-#define BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_WHITELISTS_H_
+#ifndef BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_LISTS_H_
+#define BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_LISTS_H_
```
Comment (mariospr) [inline browser/net/brave_network_audit_allowed_lists.h:7]: This should be `BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_ALLOWED_LISTS_H_` now

File: browser/net/brave_network_audit_allowed_lists.h [hunk 2]
```diff
@@ -77,12 +77,12 @@ constexpr const char* kWhitelistedUrlPrefixes[] = {
...
+constexpr const char* kAllowedUrlPatterns[] = {
     // allowed because it's url for fetching super referral's mapping table
     "https://mobile-data.s3.brave.com/superreferrer/map-table.json",
     "https://mobile-data-dev.s3.brave.software/superreferrer/map-table.json",
 };
 
 }  // namespace brave
 
-#endif  // BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_WHITELISTS_H_
+#endif  // BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_LISTS_H_
```
Comment (mariospr) [inline browser/net/brave_network_audit_allowed_lists.h:88]: This should be `BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_ALLOWED_LISTS_H_` now

**Review by mariospr — APPROVED** [review]
@rillian Thanks for changing the filename! Only thing remaining would be to fix the preprocessor guard (to `BRAVE_BROWSER_NET_BRAVE_NETWORK_AUDIT_ALLOWED_LISTS_H_`) and this would be ready to merge IMHO.

File: browser/net/brave_network_audit_allowed_lists.h (hunk 1 above)
Comment (rillian) [reply browser/net/brave_network_audit_allowed_lists.h]: Oops, thanks for the additional review!

File: browser/net/brave_network_audit_allowed_lists.h (hunk 2 above)
Comment (rillian) [reply browser/net/brave_network_audit_allowed_lists.h]: Oops, thanks for the additional review. Should be addressed now!

**Comment by UjCbFwtBayFM** [comment]
I don't see any technical/security reasons for this change, so it's a huge NO on that change.

**Pull request merged**